tests/x11/ImageMagick.pm

```diff
@@ -26,86 +26,55 @@ use testapi;
-      gradient_bilinear gradient_mesh gradient_catrom gradient_fx_linear
-      gradient_fx_x4 gradient_fx_cos gradient_fx_radial gradient_fx_spherical
-      gradient_fx_quad2 gradient_fx_angular gradient_inverse_alt
-      gradient_shepards_alt gradient_inverse_RGB gradient_inverse_RGB_Hue
-      sparse_bary_triangle sparse_barycentric sparse_bary_triangle_2
-      sparse_bary_0 sparse_bary_gradient sparse_bary_gradient_2
-      diagonal_gradient sparse_bary_two_point diagonal_gradient_2
-      sparse_bilinear sparse_bilin_0 sparse_voronoi gradient_scale
-      sparse_voronoi_ssampled gradient_math sparse_voronoi_smoothed
-      gradient_equiv sparse_voronoi_blur gradient_shifted
-      sparse_voronoi_gradient gradient_chopped sparse_shepards sparse_inverse
-      sparse_shepards_0.5 sparse_shepards_1 sparse_shepards_2 plasma_smooth
-      sparse_shepards_3 sparse_shepards_8 sparse_shepards_gray
-      rose_alpha_gradient sparse_source sparse_fill shape_edge_pixels
-      shape_edge_in_lights shape_in_lights sparse_blur_simple
-      sparse_blur_pyramid sparse_lines_near_source sparse_lines_near
-      plasma_paint plasma_emboss plasma_sharp plasma_seeded plasma_rnd1
-      plasma_rnd2 plasma_rnd3 plasma_rnd4 plasma_rnd5 random_mask random_black
-      random_white random_1 random_trans random_3 random_5 random_10 random_20
-      random_0_gray random_1_gray random_3_gray random_5_gray random_10_gray
-      random_20_gray random_0_thres random_1_thres random_3_thres random_5_thres
-      random_10_thres random_20_thres random_5_blobs ripples_1 ripples_2
-      ripples_3 ripples_4 random_enhanced ripples_4e random_sigmoidal ripples_4s
-      ripples_3e000 ripples_3e090 ripples_3e180 ripples_3e270 ripples_3.5e
-      tile_size tile_over tile_draw tile_reset tile_distort_sized offset_tile
-      offset_pattern offset_tile_fill offset_pattern_fail offset_pattern_good
-      tile_clone tile_clone_flip tile_mpr tile_mpr_reset tile_mpr_fill
-      tile_distort tile_distort_checks tile_distort_polar pattern_default
-      pattern_hexagons pattern_colored pattern_color_checks
-      pattern_color_hexagons pattern_distorted tile_mod_failure tile_mod_vpixels
-      tile_slanted_bricks tile_mod_success tile_circles tile_hexagons
-      tiled_hexagons tile_line tile_hex_lines tiled_hex_lines
+    # Prepare to run tests
+    assert_script_run "wget --quiet " . data_url('imagemagick/im_files.tar.gz') . " -O im_files.tar.gz";
+    assert_script_run "tar -xf im_files.tar.gz";
+    assert_script_run "cd im_files";
+    # Test image identification
+    my $idtf = script_output "identify test.png";
+    die "Identify failed" unless index($idtf, "test.png PNG 300x388 300x388+0+0 8-bit sRGB") != -1;
+    # Create image transformations to compare with pretransformed test images
+    my @test_imgs = (
+        "convert -fill Snow -background Green3 -strokewidth 2 -stroke Green4 -font Roboto -pointsize 256 -density 90 -size 1800x350 label:susetestlogo testlogo_t.png",
```

Comment:
Sounds like good idea, but what is all this image transformation about ?
I would expect just comparing of images.

Reply:
Thanks for checking this out Josef, the transformations are crops, resizings, etc of some base images, which then get compared to pre-transformed images with the same transformations, to evaluate weather the transformations work as expected.

Reply:
in `imagemagick/bg_script.sh` there is done resize and specific transformation for "each" image.
But here is same transformation done for all images ?
It does not look to me like same test of convert and comparing result with reference.

Reply:
It doesn't need to transform the images, because they are created with the same exact commands and from the same base images as the reference images so they are guaranteed to be the same (at least if everything is working fine).